lib/forwardanalyzer.cpp

```diff
@@ -531,7 +531,7 @@ struct ForwardTraversal {
         return updateLoop(endToken, endBlock, condTok, initTok, stepTok, true);
     }
 
-    Progress updateScope(Token* endBlock) {
+    Progress updateScope(const Token* endBlock) {
```

Comment:
This should be non-const as updateRange takes a non-const pointer. I am not sure how this even compiles.

Reply:
I see, we dont have overloads for `link(), `next()`, and `previous()` where we should ensure const-correctness.

Reply:
See #4760

Reply:
So should the new warnings be suppressed in those places?
When attempting a proper fix, there are issues like 'Should you be able to get a non-const nameToken from a const Variable', which have no good answer.

Reply:
I created #5002 to fix this issue.

> 'Should you be able to get a non-const nameToken from a const Variable', which have no good answer.

For now I just do a `const_cast`. The real fix is to add overloaded getters for this, but this needs to be propagated through symboldatabase classes.